modules/mysql/src/main/java/org/testcontainers/containers/MySQLContainerProvider.java

```diff
@@ -13,4 +30,26 @@ public boolean supports(String databaseType) {
     public JdbcDatabaseContainer newInstance(String tag) {
         return new MySQLContainer(MySQLContainer.IMAGE + ":" + tag);
     }
+    
+    @Override
+    public JdbcDatabaseContainer newInstance(ConnectionUrl url) {
+      Objects.requireNonNull(url, "Connection URL cannot be null");
+      
+      Matcher urlMatcher = MYSQL_URL_MATCHING_PATTERN.matcher(url.getUrl());
+      
+      if(!urlMatcher.matches()) {
+        //TODO: Is this necessary?
```

Comment:
Probably not necessary if the `ConnectionUrl.parseQuery()` method has been called..?

Reply:
I agree. I have removed this matcher related code.